### PR TITLE
C# useful unity & xamarin resources

### PR DESCRIPTION
**Pull request by jtattermusch**
Some useful docs to encourage collaboration.

Based on discussion from https://github.com/grpc/grpc/issues/1620 and 
https://github.com/grpc/grpc/issues/14402

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by jtattermusch** [comment]
Other useful info is here:
http://examinedself.com/cross-compile-grpc-unity/

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

File: src/csharp/experimental/build_native_ext_for_android.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+cd build
+
+# set to the location where Android SDK is installed
+ANDROID_NDK_PATH="$HOME/android-ndk-r16b"
+
+cmake ../.. \
+  -DCMAKE_SYSTEM_NAME=Android \
+  -DCMAKE_SYSTEM_VERSION=15 \
+  -DCMAKE_ANDROID_ARCH_ABI=armeabi-v7a \
+  -DCMAKE_ANDROID_NDK=${ANDROID_NDK_PATH} \
```
Comment (mehrdada) [inline src/csharp/experimental/build_native_ext_for_android.sh]: Quote "${ANDROID_NDK_PATH}"

Comment (mehrdada) [reply src/csharp/experimental/build_native_ext_for_android.sh]: (Running `shellcheck` on shellscripts is super useful btw)

**Comment by jtattermusch** [comment]
CC @zaneclaes @HI0T

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by zaneclaes** [comment]
I haven't had the chance to test this, but it looks exactly like what I implemented. I'll only note that it does not address the issue of the native Android log function not linking properly, which requires a touch more surgery to work. Otherwise, LGTM. Thanks for adding!

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by jtattermusch** [comment]
Related to Android compilation: https://github.com/grpc/grpc/pull/14193

**Comment by HI0T** [comment]
LGTM

**Pull request merged**